src/data/sparse_page_writer.h

```diff
@@ -23,9 +23,14 @@
 
 namespace xgboost {
 namespace data {
+
+template<typename T>
+struct SparsePageFormatReg;
+
 /*!
  * \brief Format specification of SparsePage.
  */
+template<typename T>
```

Comment:
Give a brief for this T in above doc please.

Reply:
Done.